File: 169-majority-element/majority-element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var maj:Int =  nums[0]
        var count = 0

        for(i in nums){
            if(i == maj) count++
            else if(count != 0) count--
            else {
                maj = i
                count = 1
            }
        } 
        return maj
    }
}

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maj = nums[0]
        appear = 0
        for num in nums:
            if num == maj:
                appear +=1
            else:
                if(appear == 0):
                    maj = num
                    appear = 1
                else:
                    appear -=1
        return maj